presto-main/src/main/java/com/facebook/presto/sql/gen/JoinCompiler.java

```diff
@@ -929,4 +1026,26 @@ public boolean equals(Object obj)
                     Objects.equals(this.sortChannel, other.sortChannel);
         }
     }
+
+    private static BytecodeNode generateArgument(
+            CallSiteBinder callSiteBinder,
+            Scope scope,
+            Type type,
+            BytecodeExpression block,
+            BytecodeExpression position)
+    {
+        Variable wasNullVariable = scope.getVariable("wasNull");
```

Comment:
Extract line 61-78 in InputReferenceCompiler into a static method. (IntelliJ can do this automatically.)

That method should be exactly the same as what you have here. Call that method directly.